Node/src/botbuilder-teams.d.ts

```diff
@@ -1279,3 +1294,327 @@ export declare class FileInfoCard implements builder.IIsAttachment {
+
+/** Implemented by classes that can be converted into an adaptive card action. */
+export interface IIsAdaptiveCardBotBuilderAction {
+
+  /** Returns the JSON object for the adaptive card action */
+  toAdaptiveCardAction(): IAdaptiveCardAction;
+}
+
+/**
+ * Adapter class for adaptive card action wrapping bot-builder actions
+ */
+export declare class AdaptiveCardBotBuilderAction implements IIsAdaptiveCardBotBuilderAction, builder.IIsCardAction {
+
+  /** Creates a new adaptive card wrapped action, from a session or wrapping an existing bot-builder action. */
+  constructor (sessionOrWrapAction?: builder.Session | builder.CardAction);
+
+  /** Type of card action. */
+  type(t: string): AdaptiveCardBotBuilderAction;
+
+  /** Title of the action. For buttons this will be the label of the button.  For tap actions this may be used for accesibility purposes or shown on hover. */
+  title(text: builder.TextType, ...args: any[]): AdaptiveCardBotBuilderAction;
+
+  /** The actions value. */
+  value(v: string): AdaptiveCardBotBuilderAction;
+
+  /** (Optional) Text for this action. */
+  text(text: builder.TextType, ...args: any[]): AdaptiveCardBotBuilderAction;
+
+  /** (Optional) text to display in the chat feed if the button is clicked. */
+  displayText(text: builder.TextType, ...args: any[]): AdaptiveCardBotBuilderAction;
+  
+  /** Returns the JSON object for the bot-builder card action. */
+  toAction(): builder.ICardAction;
+
+  /** Returns the JSON object for the adaptive card action. */
+  toAdaptiveCardAction(): IAdaptiveCardAction;
+}
+
+/**
+ * Adaptive card builder class.
+ */
+export declare class AdaptiveCard implements builder.IIsAttachment, IIsAdaptiveCard {
+
+    /** Content type of an adaptive card attachment. */
+    static readonly contentType: string;
+    
+    /** Creates a new adaptive card builder. */
+    constructor(session?: builder.Session);
+
+    /** Adaptive card root level type. Must be always 'AdaptiveCard' */
+    readonly type: 'AdaptiveCard';
+
+    /** Schema version that this card requires. If a client is lower than this version the fallbackText will be rendered. */
+    version(ver: IAdaptiveCardVersion): AdaptiveCard;
+
+    /** (Optional) An image to use as the background of the card */
+    backgroundImage(url: IAdaptiveCardBackgroundImage): AdaptiveCard;
+
+    /** (Optional) The Card Elements to show in the primary card region */
+    body(cardElements: IAdaptiveCardBody): AdaptiveCard;
+
+    /** (Optional) Specifies what should be spoken for this entire Item. This is simple text or SSML fragment. */
+    speak(text: IAdaptiveCardSpeak): AdaptiveCard;
+
+    /** (Optional) Text shown when the client doesn’t support the version specified. This can be in markdown format. */
+    fallbackText(text: IAdaptiveCardFallbackText): AdaptiveCard;
+
+    /** (Optional) The 2-letter ISO-639-1 language used in the card. Used to localize any date/time functions. */
+    lang(lang: IAdaptiveCardLang): AdaptiveCard;
```

Comment:
In the implementation we actually use the "this" return type. Why not here as well? Is there an issue with using "this" in declarations? (Asking more for my own knowledge here.)